### PR TITLE
feat(telemetry): measure duration and status

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Fixes https://github.com/netlify/cli/issues/2458

Uses a new command base class to track duration and status to avoid duplication

**- Test plan**

I tested each command manually

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: src/commands/build/index.js [hunk 1]
```diff
@@ -35,10 +37,11 @@ class BuildCommand extends Command {
 
 // Netlify Build programmatic options
 BuildCommand.flags = {
-  dry: flags.boolean({
+  dry: flagsLib.boolean({
     description: 'Dry run: show instructions without running them',
+    default: false,
```
Comment (erezrokah) [inline src/commands/build/index.js:42]: I removed the casting of `Boolean(options.dry)` in favor of a default value (without a default value not passing `--dry` results in `flags.dry` being `undefined`)

File: src/commands/dev/index.js [hunk 2]
```diff
@@ -316,6 +307,7 @@ DevCommand.flags = {
   live: flagsLib.boolean({
     char: 'l',
     description: 'start a public live session',
+    default: false,
```
Comment (erezrokah) [inline src/commands/dev/index.js:310]: Same as before I removed `flags.live || false` in favor of a default value

File: src/commands/functions/build.js [hunk 3]
```diff
@@ -1,30 +1,24 @@
...
-    })
-
     const { config } = this.netlify
 
     const src = flags.src || config.build.functionsSource
     const dst = getFunctionsDir({ flags, config })
 
     if (src === dst) {
       this.log(`${NETLIFYDEVERR} Source and destination for function build can't be the same`)
-      process.exit(1)
```
Comment (erezrokah) [inline src/commands/functions/build.js:27]: This prevents from the analytics to be reported in the `finally` block

File: src/utils/telemetry/tracked-command.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  async init() {
+    this.analytics = { startTime: hrtime.bigint() }
+    await super.init()
+  }
+
+  setAnalyticsPayload(payload) {
+    this.analytics = { ...this.analytics, payload }
+  }
+
+  async finally(error) {
```
Comment (erezrokah) [inline src/utils/telemetry/tracked-command.js:38]: For this to work well we need commands to use `this.error` or `this.exit` which most commands do.
Some areas of the codebase use `process.exit` which we should change.
I think we can separate it into a dedicated PR as it will require some refactoring.

Also, we would need to verify that commands report errors properly, but we can do that gradually.

File: src/utils/telemetry/tracked-command.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  if (error === undefined || isSafeError(error)) {
+    return 'success'
+  }
+
+  return 'error'
+}
+
+class TrackedCommand extends Command {
+  async init() {
+    this.analytics = { startTime: hrtime.bigint() }
```
Comment (erezrokah) [inline src/utils/telemetry/tracked-command.js:30]: We could do this as a class properly to avoid someone extending and forgetting to call `super.init`.

However, based on [this code](https://github.com/oclif/command/blob/adc0b266777b515a26365d31c0e3a926b68d0c72/src/command.ts#L145) I think `init` should always be called, and it seems the correct place to start measuring command time.

We could also move it to `run` but then we would have to call `super.run` in every command which doesn't feel correct.

**Review by JGAntunes — APPROVED** [review]
🚀 Really looking forward to this! 😄
